src/lib/ext-activitylog.js

```diff
@@ -65,11 +240,34 @@ class Controller {
     const existingLogs = await this.getExistingLogs();
 
     if (existingLogs.length) {
-      this.model.addNewLogs(existingLogs);
-      this.view.addTableRows(this.model.logs);
+      this.handleNewLogs(existingLogs);
+    }
+  }
+
+  handleNewLogs(logs) {
+    for (const log of logs) {
+      if (log.viewType === undefined) {
+        log.viewType = 'undefined';
```

Comment:
Don't modify the log entry from the controller. The logs should be considered read-only.

Reply:
Okay, done :-)